pkg/asset/machines/azure/zones.go

```diff
@@ -0,0 +1,57 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
+	azureutil "github.com/openshift/installer/pkg/asset/installconfig/azure"
+)
+
+// AvailabilityZones retrieves a list of availability zones for the given region and instance type.
+func AvailabilityZones(region string, instanceType string) ([]string, error) {
+	skusClient, err := skusClient()
+	if err != nil {
+		return nil, err
+	}
+	zones, err := fetchAvailabilityZones(skusClient, region, instanceType)
+	if err != nil {
+		return nil, fmt.Errorf("cannot fetch availability zones: %v", err)
+	}
+	return zones, nil
+}
+
+func skusClient() (client *compute.ResourceSkusClient, err error) {
+	ssn, err := azureutil.GetSession()
+	if err != nil {
+		return nil, err
+	}
+
+	skusClient := compute.NewResourceSkusClient(ssn.Credentials.SubscriptionID)
+	skusClient.Authorizer = ssn.Authorizer
+	return &skusClient, nil
+}
+
+func fetchAvailabilityZones(client *compute.ResourceSkusClient, region string, instanceType string) ([]string, error) {
+	var zones []string
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+
+	res, err := client.List(ctx)
```

Comment:
Might need to use ListComplete (since List provides only the first page, which may not be complete)

Example: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/9b553da079817f83fe7659696d5a88b60724463d/pkg/cloud/azure/services/availabilityzones/availabilityzones.go#L42

Probably require same fix in openshift branch of cluster api azure

Reply:
here's an example of using list and iterating of pages.
https://github.com/openshift/installer/blob/a969950ec2a63eab8b0bb8017a3b489211bbae00/pkg/destroy/azure/azure.go#L109-L119